google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/facet/StackdriverPanel.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.debugger.facet;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+
+import com.intellij.facet.ui.FacetEditorContext;
+import com.intellij.facet.ui.FacetEditorTab;
+import com.intellij.openapi.options.ConfigurationException;
+
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+
+/**
+ * Stackdriver's facet user interface.
+ */
+public class StackdriverPanel extends FacetEditorTab {
+
+  private JCheckBox generateSourceContext;
+  private JPanel stackdriverPanel;
+  private JCheckBox ignoreErrors;
+  private FacetEditorContext editorContext;
+  private StackdriverFacetConfiguration configuration;
+
+  public StackdriverPanel() {
+    generateSourceContext.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        ignoreErrors.setEnabled(!ignoreErrors.isEnabled());
```

Comment:
The problem here is that when you unselect the "Generate source context files" checkbox, `generateSourceContext.isSelected()` only returns `false` after this method returns.

Instead of hacking around that and using an unintuitive `ignoreErrors.setEnabled(!generateSourceContext.isSelected())`, I went for this. These checkboxes are correctly initialized everywhere, so toggling should always work here. I tested a bit and it works as expected.